# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/BasicEvents.h
# Commit: aefcf10217ab
# Full Hash: aefcf10217ab75ebb5ae82cf4fef71a16db897fa
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-12-03 16:21:08
# Regressor Bug: 1680669
# File Overlap Count: 1
# Description:
#   Bug 1680669 - part 3: Add a utility method to `WidgetEvent` to check whether the corresponding DOM event supports coordinates r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D222725
# ==============================================================================

diff -r edde7fe51b4b -r aefcf10217ab widget/BasicEvents.h
--- a/widget/BasicEvents.h	Tue Dec 03 07:09:44 2024 +0000
+++ b/widget/BasicEvents.h	Tue Dec 03 07:09:49 2024 +0000
@@ -740,6 +740,24 @@
   inline bool IsReservedByChrome() const { return mFlags.IsReservedByChrome(); }
 
   /**
+   * Return true if the corresponding DOM event supports screen(X|Y), etc.
+   */
+  [[nodiscard]] inline bool DOMEventSupportsCoords() const {
+    switch (mClass) {
+      case eMouseEventClass:
+      case eMouseScrollEventClass:
+      case eWheelEventClass:
+      case eTouchEventClass:
+      case eDragEventClass:
+      case ePointerEventClass:
+      case eSimpleGestureEventClass:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  /**
    * Utils for checking event types
    */
 
